source/ut_utils.pks

```diff
@@ -21,24 +21,25 @@ create or replace package ut_utils is
   tr_failure_char            constant varchar2(7) := 'Failure'; -- one or more asserts failed
   tr_error_char              constant varchar2(5) := 'Error'; -- exception was raised
 
-  
   /*
     Constants: Rollback type for ut_test_object
   */
   gc_rollback_auto           constant number(1) := 0; -- rollback after each test and suite
   gc_rollback_manual         constant number(1) := 1; -- leave transaction control manual
   --gc_rollback_on_error       constant number(1) := 2; -- rollback tests only on error
-  
+
   ex_unsopported_rollback_type exception;
   pragma exception_init(ex_unsopported_rollback_type, -20200);
 
-
-  gc_max_sring_length        constant integer := 4000;
-  gc_more_data_string        constant varchar2(5) := '[...]';
-  gc_overflow_substr_len     constant integer := gc_max_sring_length - length(gc_more_data_string);
-  gc_number_format           constant varchar2(100) := 'TM9';
-  gc_date_format             constant varchar2(100) := 'yyyy-mm-dd hh24:mi:ss';
-  gc_timestamp_format        constant varchar2(100) := 'yyyy-mm-dd hh24:mi:ssxff';
+  gc_max_output_string_length constant integer := 4000;
+  gc_max_input_string_length  constant integer := gc_max_output_string_length - 2; --we need to remove 2 chars for quotes around string
+  gc_more_data_string         constant varchar2(5) := '[...]';
+  gc_overflow_substr_len      constant integer := gc_max_input_string_length - length(gc_more_data_string);
+  gc_number_format            constant varchar2(100) := 'TM9';
+  gc_date_format              constant varchar2(100) := 'yyyy-mm-dd"T"hh24:mi:ss';
+  gc_timestamp_format         constant varchar2(100) := 'yyyy-mm-dd"T"hh24:mi:ssxff';
```

Comment:
I get full precision without using `ff9`.
I'll touch that topic while working on XML date format issue.